pkgconfig/mbedtls.pc.in

```diff
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+includedir=@PKGCONFIG_INCLUDEDIR@
+libdir=@PKGCONFIG_LIBDIR@
+
+Name: @PROJECT_NAME@
+Description: @PKGCONFIG_PROJECT_DESCRIPTION@
+URL: @PKGCONFIG_PROJECT_HOMEPAGE_URL@
+Version: @PROJECT_VERSION@
+Requires.private: mbedcrypto mbedx509
```

Comment:
Late feedback: shouldn't this be `Requires` instead of `Requires.private`? The CMake code, for instance, has a `PUBLIC` dependency on `mbedx509`.

https://github.com/Mbed-TLS/mbedtls/blob/30978ec6507c0a0376ae041fa333a2f24b0c09b8/library/CMakeLists.txt#L314-L320

Reply:
That depends on the following question. If a program links with mbedtls _only_ dynamically, does it need to link with mbedcrypto and mbedx509 directly?

Reply:
Requires.private is correct here. Those libraries are linked to the shared object not the requestor.

I think this explains it better than me:
- https://stackoverflow.com/questions/45441250/how-does-the-linking-process-differ-when-using-requires-vs-requires-private-in-p

Reply:
> That depends on the following question. If a program links with mbedtls _only_ dynamically, does it need to link with mbedcrypto and mbedx509 directly?

I was under the impression that this was the case, but no, `Requires.private` seems appropriate. `mbedtls/ssl.h` includes `mbedtls/x509_crt.h`, which is part of `mbedx509`, so the user ends up needing the appropriate _compiler_ flags for `mbedx509` too, but unless it uses x509 functions it does not need its _linking_ flags. Due to [multiple reasons](https://gitlab.freedesktop.org/pkg-config/pkg-config/-/issues/7), pkg-config's `Requires.private` propagates the `Cflags` when doing shared builds, but not the `Libs`, so this is exactly what we need.

In short: the pkg-config files are ok, but `mebdtls`' `PUBLIC` dependency on `mbedx509` in CMake is probably wrong.